Xamarin.Essentials/Browser/Browser.shared.cs

```diff
@@ -35,9 +54,65 @@ internal static Uri EscapeUri(Uri uri)
         }
     }
 
+    public class BrowserLaunchOptions : IEquatable<BrowserLaunchOptions>
+    {
+        public BrowserLaunchOptions(Color? prefferedControlColor, Color? preferredBackgroundColor, Color? preferredTitleColor, BrowserLaunchMode launchMode, BrowserTitleMode shouldShowTitle)
+        {
+            PrefferedControlColor = prefferedControlColor;
+            PreferredBackgroundColor = preferredBackgroundColor;
+            PreferredTitleColor = preferredTitleColor;
+            LaunchMode = launchMode;
+            TitleMode = shouldShowTitle;
+        }
+
+        public BrowserLaunchOptions()
+        {
+            LaunchMode = BrowserLaunchMode.Default;
+        }
+
+        public BrowserLaunchOptions(BrowserLaunchMode mode)
+        {
+            LaunchMode = mode;
+        }
+
+        public Color? PreferredTitleColor { get; set; }
+
+        public Color? PreferredBackgroundColor { get; set; }
+
+        public Color? PrefferedControlColor { get; set; }
+
+        public BrowserLaunchMode LaunchMode { get; set; }
+
+        public BrowserTitleMode TitleMode { get; set; }
+
+        public static bool operator ==(BrowserLaunchOptions lhs, BrowserLaunchOptions rhs) => lhs.Equals(rhs);
+
+        public static bool operator !=(BrowserLaunchOptions lhs, BrowserLaunchOptions rhs) => lhs.Equals(rhs);
+
+        public override bool Equals(object other) => other is BrowserLaunchOptions options && other != null && Equals(options);
+
+        public bool Equals(BrowserLaunchOptions other) => other != null && PreferredTitleColor.Equals(other.PreferredTitleColor)
+                                                          && PreferredBackgroundColor.Equals(
+                                                              other.PreferredBackgroundColor)
+                                                          && PrefferedControlColor.Equals(other.PrefferedControlColor)
+                                                          && LaunchMode.Equals(other.LaunchMode)
+                                                          && TitleMode.Equals(other.TitleMode);
+
+        public override int GetHashCode() => (PreferredBackgroundColor, PreferredBackgroundColor, PrefferedControlColor,
+            TitleMode, LaunchMode).GetHashCode();
+    }
+
+    public enum BrowserTitleMode
+    {
+        Default = 0,
+        Show = 1,
+        Hide = 2
+    }
+
     public enum BrowserLaunchMode
     {
-        SystemPreferred = 0,
-        External = 1,
+        Default = 0,
```

Comment:
@jamesmontemagno Do we want a "default" option here? If follows the rules, but is this maybe not useful? I think our original idea that "Default" === "SystemPreferred", but we used a different name to help discoverability? Not sure about this one...

Reply:
I don't think we wan't default here.

SystemPreferred != Default as Default would be launching URI. This is a special case in general and should be the default. I think that we should pass in the LaunchMode as a parameter and LaunchOptions as another. I would hate to create another class just to do systempreferred or external if I don't need any of the options.

Reply:
If I might refer to the discussion we had about this being a value or reference type, if this were a value type the object creation and destruction would be negligable. Especially when we consider the use case, this will not be called frequently. Otherwise I can revert to the three parameter style I adopted originally for the internal implementation and just pass null.